src/NuGet.Clients/PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -249,18 +262,8 @@ protected void SelectVersion()
                 return;
             }
 
-            DisplayVersion versionToSelect = _versions
-                .Where(v => v != null && v.Version.Equals(_searchResultPackage.Version))
-                .FirstOrDefault();
-            if (versionToSelect == null)
-            {
-                versionToSelect = _versions[0];
-            }
-
-            if (versionToSelect != null)
-            {
-                SelectedVersion = versionToSelect;
-            }
+            // it should always select the top version from versions list to install or update
+            SelectedVersion = _versions[0];
```

Comment:
It already checks for versions available. With disabled version, it just needs to further check for valid versions which i'm adding with new changes.